pkgs/servers/pleroma/default.nix

```diff
@@ -0,0 +1,206 @@
+        version = "1.0.0";
+
+        src = fetchFromGitLab {
+          domain = "git.pleroma.social";
+          group = "pleroma";
+          owner = "elixir-libraries";
+          repo = "majic";
+          rev = "289cda1b6d0d70ccb2ba508a2b0bd24638db2880";
+          sha256 = "15605lsdd74bmsp5z96f76ihn7m2g3p1hjbhs2x7v7309n1k108n";
+        };
+        patchPhase = ''
+          sed 's/erlang.now/erlang.time/g' -i lib/majic/server.ex
+        '';
+        buildInputs = [ file ];
+
+        beamDeps = with final; [ nimble_pool mime plug elixir_make ];
+      };
+      crypt = beamPackages.buildRebar3 rec {
+        name = "crypt";
+        version = "0.4.3";
+
+        src = fetchFromGitLab {
+          domain = "git.pleroma.social";
+          group = "pleroma";
+          owner = "elixir-libraries";
+          repo = "crypt";
+          rev = "cf2aa3f11632e8b0634810a15b3e612c7526f6a3";
+          sha256 = "0fnzljxy9pwabh1nzx0vawn131d5pdfb0p98kvpkqs441jr0ii73";
+        };
+
+        postInstall = "mv $out/lib/erlang/lib/crypt-${version}/priv/{source,crypt}.so";
+
+        beamDeps = with final; [ elixir_make ];
+      };
+      web_push_encryption = beamPackages.buildMix rec {
+        name = "web_push_encryption";
+        version = "0.3.0";
+        src = fetchFromGitHub {
+          owner = "lanodan";
+          repo = "elixir-web-push-encryption";
+          rev = "026a043037a89db4da8f07560bc8f9c68bcf0cc0";
+          sha256 = "0a4x6njqp8v579bc965c9ipsr1z3klrc0pvgj8x1xf69r77gs6sj";
+        };
+        beamDeps = with final; [ httpoison jose ];
+      };
+
+      # Some additional build inputs and build fixes
+      http_signatures = prev.http_signatures.override {
+        patchPhase = ''
+          sed 's/:logger/&, :public_key/' -i mix.exs
+        '';
+      };
+      fast_html = prev.fast_html.override {
+        nativeBuildInputs = [ cmake ];
+        dontUseCmakeConfigure = true;
+      };
+      syslog = prev.syslog.override {
+        buildPlugins = with beamPackages; [ pc ];
+      };
+
+      # This is wrongly identified as a Rebar3 dep
+      jose = beamPackages.buildMix rec {
+        name = "jose";
+        version = "1.11.1";
+
+        src = beamPackages.fetchHex {
+          pkg = "${name}";
+          version = "${version}";
+          sha256 = "1i8szzpmiqc7xdv0lp38ng9fild7c5182b4pzkx4qbydnfgnr3q7";
+        };
+
+        beamDeps = [];
+      };
+
+      # This needs a different version (1.0.14 -> 1.0.18) to build properly with
+      # our Erlang/OTP version.
+      eimp = beamPackages.buildRebar3 rec {
+        name = "eimp";
+        version = "1.0.18";
+
+        src = beamPackages.fetchHex {
+          pkg = "${name}";
+          version = "${version}";
+          sha256 = "0fnx2pm1n2m0zs2skivv43s42hrgpq9i143p9mngw9f3swjqpxvx";
+        };
+
+        patchPhase = ''
+          echo '{plugins, [ { pc, { git, "git://github.com/blt/port_compiler.git", {tag, "v${beamPackages.pc.version}"}}} ]}.' >> rebar.config
```

Comment:
@dlesl just in case you have something to add on this

Reply:
It looks like this library uses `rebar`, not `rebar3`, so I think injecting this here is a reasonable thing to do. However it is not necessary to specify a version, `{plugins, [pc]}` should do -- `rebar3WithPlugins` will not try to fetch it from `hex.pm`

Reply:
Thanks for the hint, will implement